tests/mmm/test_utility.py

```diff
@@ -0,0 +1,376 @@
+#   Copyright 2024 The PyMC Labs Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+import numpy as np
+import pymc as pm
+import pytensor.tensor as pt
+import pytest
+from pytensor import function
+
+from pymc_marketing.mmm.utility import (
+    _calculate_roas_distribution_for_allocation,
+    _compute_quantile,
+    _covariance_matrix,
+    adjusted_value_at_risk_score,
+    average_response,
+    conditional_value_at_risk,
+    mean_tightness_score,
+    portfolio_entropy,
+    raroc,
+    sharpe_ratio,
+    tail_distance,
+    value_at_risk,
+)
+
+rng: np.random.Generator = np.random.default_rng(seed=42)
+
+EXPECTED_RESULTS = {
```

Comment:
where are these values coming from?

Reply:
I ran a notebook to check what would be the results for each function given the parameters, this test validates they return those values every time, meaning, function behavior its not changing and return consistent results.